astropy/wcs/src/unit_list_proxy.c

```diff
@@ -186,6 +187,59 @@ PyUnitListProxy_getitem(
   return result;
 }
 
+static PyObject *
+PyUnitListProxy_richcmp(
+    PyUnitListProxy *a,
+    PyUnitListProxy *b,
+    int op) {
+
+  int equal;
+  int status;
+  struct PyUnitListProxy *ax;
+  struct PyUnitListProxy *bx;
+   
+  
+  if ((op == Py_EQ || op == Py_NE) &&
+      PyObject_TypeCheck(b, &PyUnitListProxyType)) {
+      ax = &((PyUnitListProxy *)a)->pyobject;
+      bx = &((PyUnitListProxy *)b)->pyobject;
+          
+    
+    if (!PyObject_RichCompareBool(a->unit_class, b->unit_class,Py_EQ)||
+        a->size != b->size||
+        strncmp(a->array, b->array, 72) ) {
```

Comment:
The `72` is probably hard-coded somewhere else, could you use a `#define` here so the values never diverge?

Reply:
Except of creating #define can i create a function to compare which will return 1 for true and 0 false  because a loop will be needed to compare (*)[72]

Reply:
Sorry, I don't understand what you mean.